docs/ref/models/constraints.txt

```diff
@@ -152,6 +152,12 @@ For example, ``UniqueConstraint(fields=['room', 'date'],
 name='unique_booking')`` ensures each room can only be booked once for each
 date.
 
+If a :class:`~django.core.exceptions.ValidationError` is raised during a
+model's :meth:`validate_constraints() <Model.validate_constraints>` it will be
+classified as a field-error for ``fields`` listing only a single field;
+for ``fields`` listing multiple fields it will be classified as a
+:attr:`non-field-error <django.core.exceptions.NON_FIELD_ERRORS>`.
```

Comment:
I'd move this to the `violation_error_message` and simplified a bit, e.g.
```diff
diff --git a/docs/ref/models/constraints.txt b/docs/ref/models/constraints.txt
index d0a9a017dc..0ce183ef77 100644
--- a/docs/ref/models/constraints.txt
+++ b/docs/ref/models/constraints.txt
@@ -239,7 +239,9 @@ creates a unique index on ``username`` using ``varchar_pattern_ops``.
 
 The error message used when ``ValidationError`` is raised during
 :ref:`model validation <validating-objects>`. Defaults to
-:attr:`.BaseConstraint.violation_error_message`.
+:attr:`.BaseConstraint.violation_error_message`. Such errors are
+field errors for single field constraints, and non-field errors
+for multiple field constraints.
 
 This message is *not used* for :class:`UniqueConstraint`\s with
 :attr:`~UniqueConstraint.fields` and without a

```

Reply:
@felixxm point taken and it's a good idea … this only affects `UniqueConstraint` + `fields` so I'd need to at least clarify this and refer to the fields section 👍